### PR TITLE
Upgrade to Ember 3

### PR DESCRIPTION
**Pull request by pcambra**
Followed this amazing guide to do so https://dockyard.com/blog/2018/01/18/test-helpers-the-next-generation

Not sure if you'd like to keep both 2 and 3 running at the same time or just create a new release/branch for Ember 3 version.

### CODE REVIEW

**Comment by pcambra** [comment]
Fixed one of the tests but not sure what could be wrong with the remainder, any clues?

File: addon-test-support/index.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      // get the page before is our target
+      // remove that one also, since it's readded by transition
+      const lastPage = this.history.pop();
+      assert('A transition must be recognized after `setupBrowserNavigationButtons` was called.', lastPage);
+      return this.router.transitionTo(lastPage);
+    },
+    goForward() {
+      const nextPage = this.forward.pop();
+      assert('backButton must be used atleast one time before forwardButton could be used..', nextPage);
+      this.router.transitionTo(nextPage);
```
Comment (jelhan) [inline addon-test-support/index.js]: You are missing a `return` here.

**Comment by jelhan** [comment]
Thanks for the amazing work! Will try to do a full review as soon as possible.

**Comment by pcambra** [comment]
🎉

File: package.json [hunk 2]
```diff
@@ -14,31 +14,37 @@
...
-    "ember-cli": "~2.16.2",
+    "ember-cli": "~3.1.4",
     "ember-cli-dependency-checker": "^2.0.0",
-    "ember-cli-eslint": "^4.0.0",
+    "ember-cli-eslint": "^4.2.1",
     "ember-cli-htmlbars": "^2.0.1",
     "ember-cli-htmlbars-inline-precompile": "^1.0.0",
     "ember-cli-inject-live-reload": "^1.4.1",
-    "ember-cli-qunit": "^4.0.0",
-    "ember-cli-release": "^1.0.0-beta.2",
```
Comment (jelhan) [inline package.json]: ember-cli-release dependency should not be removed.

**Comment by jelhan** [comment]
Do you have time to address the last remaining issues? I can merge and release afterwards.

File: package.json (hunk 2 above)
Comment (pcambra) [reply package.json]: Ah, missed that one when doing ember init 👍

**Comment by pcambra** [comment]
I'm too used to npm, didn't notice the yarn.lock file :)

**Comment by pcambra** [comment]
Let me know if there's anything else to update, I rather leave README updates to you, if you don't mind :)

**Comment by jelhan** [comment]
@pcambra Could it be that you are using an outdated yarn version? If I'm running `yarn install` with latest stable release (v1.7.0), I'm seeing some small changes to `yarn.lock`.

**Comment by jelhan** [comment]
Released v0.1.0 including this PR and some other improvements.

**Comment by pcambra** [comment]
Weird, I do have that same version, I think did remove the yarn.lock before generating it though

```
yarn --version
1.7.0
```

Thanks!

**Pull request merged**